store/schemagraph.go

```diff
@@ -6,16 +6,50 @@ import (
 	"github.com/valocode/bubbly/api/core"
 )
 
-// schemaGraph creates a graph from the bubbly schema
+// relType describes the relationship type of a directed edge from a --> b
+type relType int
+
+// FIXME: ofr what's the difference between `oneToOne` and `belongsTo`, please?
+const (
+	oneToOne relType = iota
+	oneToMany
+	belongsTo
+)
+
+// schemaNode represents a node in the schema graph.
+// A node is a wrapper around core.Table with the edges for explicit
+// relationships to other nodes (and therefore tables)
+type schemaNode struct {
+	table *core.Table
+	edges schemaPath
+}
+
+// nodeRefMap maps node names to the corresponding structures of type node
+type nodeRefMap map[string]*schemaNode
+
+// schemaNodes is a list of graph nodes
+type schemaNodes []*schemaNode
+
+// schemaEdge represents an edge in the graph
+type schemaEdge struct {
+	node *schemaNode
+	rel  relType
+}
+
+// schemaPath is a list graph edges
+type schemaPath []*schemaEdge
+
+// schemaGraph represents a graph created from the bubbly schema.
 type schemaGraph struct {
-	nodes []*schemaNode
+	nodes schemaNodes
 	// nodeIndex stores an index to the nodes using the schema table name.
 	// This is probably not the best for performance, but our schemas probably
 	// will not become huge and this is a great convienice for consumers of the
 	// graph, to not have to traverse the graph to find a node
-	nodeIndex map[string]*schemaNode
+	nodeIndex nodeRefMap
 }
 
+// traverse ???
```

Comment:
traverse traverses the schemaGraph and calls the callback function for each node